src/simplify_reshapes.cpp

```diff
@@ -806,6 +806,31 @@ struct find_transpose_slice
     }
 };
 
+// replace scalar (multibroadcast->unsqueeze) with just an updated (multibroadcast)
+struct find_scalar_multibroadcast_unsqueeze
+{
+    auto matcher() const
+    {
+        auto contiguous = match::name("contiguous");
+        auto mbr        = match::name("multibroadcast");
```

Comment:
This should check for scalar in the matcher. You can add a `scalar_shape` matcher to matcher.hpp(put it next to `broadcast_shape` matcher):

```cpp
MIGRAPHX_PRED_MATCHER(scalar_shape, instruction_ref ins)
{
    return ins->get_shape().scalar();
}
```

And then do `match::name("multibroadcast")(scalar_shape())`.